src/datahike/index/hitchhiker_tree/upsert.cljc

```diff
@@ -43,8 +44,9 @@
       ;; '-' means it is retracted and 'nt' is the current transaction time.
       [a b c (- nt)])))
 
-(defrecord temporal-UpsertOp [key]
+(defrecord temporal-UpsertOp [key ts]
```

Comment:
I think it would have been better to name this `TemporalUpsertOp`. We can rename it, you just need to add a read-handler that automatically transforms the old records to the new name. But we can also do so later.